### ISSUE TITLE
How to read permission grants?

### ISSUE THREAD

**Issue by TowhidKashem** [agencyenterprise/react-native-health#35]
Is it possible to tell if the user has granted permissions for a particular write?

I'm not sure if this library is built on top of `rn-apple-healthkit` or not but that lib seems to have a function for this:

* https://github.com/terrillo/rn-apple-healthkit/issues/82
* https://github.com/terrillo/rn-apple-healthkit/blob/master/docs/authorizationStatusForType().md

Could we get something similar here?

**Comment by seidbar** [comment]
It should be the same here: `https://github.com/agencyenterprise/react-native-health/blob/master/docs/authorizationStatusForType.md`

**Comment by TowhidKashem** [comment]
@seidbar I tried that before opening the issue. That function doesn't seem to exist in this lib. I did a search for `authorizationStatusForType` in `node_modules/react-native-health` and it turned out 0 results. Running the code in the link above results in the expected error:

<img width="772" alt="Screen Shot 2021-01-02 at 9 52 36 PM" src="https://user-images.githubusercontent.com/1139927/103470800-23954780-4d45-11eb-83a8-80d63fb854f3.png">

**Comment by vishalj** [comment]
This is now fixed with https://github.com/agencyenterprise/react-native-health/blob/master/docs/getAuthStatus.md
Although there is a bug I filled here https://github.com/agencyenterprise/react-native-health/issues/42 that has been now introduced

**Comment by TowhidKashem** [comment]
hey @vishalj that's great to hear, I tried it but am not seeing what the expected result. So I'm using the library to write to `Mindful Minutes`. I allowed the app access to `MindfulSession` writes and confirmed I can write to it, for example if I go to my settings > health data I see 10 minutes there for today's Mindful Minutes.

Then I tried to read permissions like:

```
    const permissions = {
      permissions: {
        read: [AppleHealthKit.Constants.Permissions.MindfulSession],
        write: [AppleHealthKit.Constants.Permissions.MindfulSession]
      }
    } as HealthKitPermissions;

    AppleHealthKit.getAuthStatus(permissions, (err, results) => {
      console.log(err, results);
    });
```

And I get:

```
permissions: {
  read: [0],
  write: [0]
}
```

Shouldn't I be getting `write[2]`?

**Comment by vishalj** [comment]
@TowhidKashem I am seeing write[2,2,2,2] for my categories 
```
permissions: {
    read: [
      AppleHealthKit.Constants.Permissions.Workout,
      AppleHealthKit.Constants.Permissions.ActiveEnergyBurned,
      AppleHealthKit.Constants.Permissions.BasalEnergyBurned,
      AppleHealthKit.Constants.Permissions.HeartRate,
    ],
    write: [
      AppleHealthKit.Constants.Permissions.Workout,
      AppleHealthKit.Constants.Permissions.ActiveEnergyBurned,
      AppleHealthKit.Constants.Permissions.BasalEnergyBurned,
      AppleHealthKit.Constants.Permissions.HeartRate,
    ],
  }```

**Comment by vishalj** [comment]
@TowhidKashem  Just in case you have seen this issue before, You are right I am running into the issue as well. In the beginning it returns with all non determined values(0), but after` AppleHealthKit.initHealthKit` is called, The correct values are returned.

@NS-BOBBY-C Tagging you as you fixed this functionality and maybe have more context, if that is how it was implemented.

**Comment by vishalj** [comment]
Just went through the code, we need to add `self.healthStore = [[HKHealthStore alloc] init];` in getAuthorizationStatus as well, I think. Haven't worked on the native side so will need to look if that's all that's needed, or if we need to check wether or not the healthstore has been initialized before calling it again.

**Comment by Macelai** [comment]
Hey, this issue seems to be fixed by #44, I'm getting read/write equal 2. I'll close this issue, but feel free to open a new one if something is missing.

**Comment by jaroslav009** [comment]
> Hey, this issue seems to be fixed by #44, I'm getting read/write equal 2. I'll close this issue, but feel free to open a new one if something is missing.

It's not closed I'm still getting 1 in AppleHealthKit.Constants.Permissions.HeartRate and it could be turned on or turned off it'll be always 1
